packages/api/src/command/medical/cq-directory/create-cq-directory-entry.ts

```diff
@@ -1,32 +1,114 @@
-export type CQOrganizationCreateResponse = { org: CQDirectoryEntry } & {
-  updated: boolean;
+const QueryResultSchema = z.array(
+  z.object({
+    point: z.string(),
+  })
+);
+
+const sqlDBCreds = Config.getDBCreds();
+const dbCreds = JSON.parse(sqlDBCreds);
+
+const sequelize = new Sequelize(dbCreds.dbname, dbCreds.username, dbCreds.password, {
+  host: dbCreds.host,
+  port: dbCreds.port,
+  dialect: dbCreds.engine,
+});
+
+export type CQOrganizationCreateResponse = { org: CQDirectoryEntry } & { updated: boolean };
+export type CQOrganizatioBulkCreateResponse = { updated: number; added: number };
+export type CQDirectoryEntryDataWithUpdateAndId = CQDirectoryEntryData & {
+  update?: boolean;
+  new?: boolean;
 };
 
 export const createOrUpdateCQDirectoryEntry = async (
   orgData: CQDirectoryEntryData
 ): Promise<CQOrganizationCreateResponse> => {
-  // ensure we never create more than one entry per cq org
   const existingOrg = await getCQDirectoryEntry({ oid: orgData.oid });
   if (existingOrg) {
     const updOrg = await updateCQDirectoryEntry({ ...orgData, id: existingOrg.id });
     return { org: updOrg, updated: true };
   }
 
-  const org = await createDirectoryDirectoryEntry(orgData);
+  const org = await createCQDirectoryEntry(orgData);
   return { org, updated: false };
 };
 
-async function createDirectoryDirectoryEntry(
-  orgData: CQDirectoryEntryData
-): Promise<CQDirectoryEntry> {
-  return await CQDirectoryEntryModel.create({
-    id: uuidv7(),
-    ...orgData,
+const createCQDirectoryEntry = async (orgData: CQDirectoryEntryData): Promise<CQDirectoryEntry> => {
+  return await CQDirectoryEntryModel.create(orgData);
+};
+
+export const createOrUpdateCQDirectoryEntries = async (
+  orgDataArray: CQDirectoryEntryDataWithUpdateAndId[]
+): Promise<CQOrganizatioBulkCreateResponse> => {
+  const oids = orgDataArray.map(data => data.oid);
+  const existingEntries = await getCQDirectoryEntriesIdsAndLastUpdated(oids); // TODO: could parallelize this
+
+  orgDataArray.forEach(orgData => {
+    const existingEntry = existingEntries.find(entry => entry.oid === orgData.oid);
+    if (existingEntry) {
+      const latestVersion = existingEntry.lastUpdated === orgData.lastUpdated;
+      if (!latestVersion) {
+        orgData.id = existingEntry.id;
+        orgData.update = true;
+      }
+    } else {
+      orgData.new = true;
+    }
   });
+
+  const newEntries = orgDataArray.filter(org => org.new);
+  const numNewEntries = newEntries.length;
+
+  const updEntries = orgDataArray.filter(org => org.update);
+  const numUpdEntries = updEntries.length;
+
+  if (numNewEntries) await createCQDirectoryEntries(newEntries);
+  if (numUpdEntries) await updateCQDirectoryEntries(updEntries);
+
+  return { added: numNewEntries, updated: numUpdEntries };
+};
+
+async function computeEarthPoint(orgData: CQDirectoryEntryData): Promise<string | undefined> {
+  if (orgData.lat && orgData.lon) {
+    const query = "SELECT ll_to_earth(:lat, :lon) as point";
```

Comment:
use sequelize orm instead of raw sql querries. So something like this instead:
`const pointQueryResult = await CQDirectoryEntryModel.findOne({
      attributes: [[sequelize.fn('ll_to_earth', orgData.lat, orgData.lon), 'point']],
      raw: true,
`

Reply:
I'd consider the pros and cons here. I'm 100% for using Sequelize's ORM goodies in the vast majority of cases, especially to have type safety. But, if we don't get type safety from it and it makes it harder to code, I'd go with pure SQL for very specific applications/use cases.